### PR TITLE
Make asset download feature usable by both "jobs post" and "isos post"

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

**Comment by mudler** [comment]
Please rebase (so also when merging history is not mixed), otherwise looks good

**Comment by okurz** [comment]
rebased

File: lib/OpenQA/Utils.pm [hunk 1]
```diff
@@ -715,6 +718,77 @@ sub check_download_whitelist {
...
+        # Find where we should download the file to
+        my $fullpath = locate_asset($assettype, $filename, mustexist => 0);
+
+        unless (-s $fullpath) {
+            # if the file doesn't exist, add the url/target path and extraction
+            # flag as a key/value pair to the %downloads hash
+            $downloads{$url} = [$fullpath, $do_extract];
+        }
+    }
+    return %downloads;
```
Comment (Martchus) [inline lib/OpenQA/Utils.pm]: Maybe return a reference to prevent copying the hash?

**Comment by Martchus** [comment]
I re-triggered the tests and the negative results don't look like the usual random failures we currently see.

File: lib/OpenQA/Utils.pm (hunk 1 above)
Comment (okurz) [reply lib/OpenQA/Utils.pm]: done

**Comment by okurz** [comment]
I am sure the tests fail because of my changes but since about 1 year I struggle to execute openQA tests or read the results. Before major changes to logging and the test infrastructure there was the time when I could find the "prove" output and see what is going on. Now in travis I only find loads of log output and I do not know if they refer to successful or failed tests

**Comment by coolo** [comment]
you need a trained eye to spot:
```
#   Failed test 'Compile test for lib/OpenQA/WebAPI/Controller/API/V1/Iso.pm'
#   at ./t/01-compile-check-all.t line 30.

**Comment by coolo** [comment]
the most annoying one is from coverage library

**Comment by okurz** [comment]
fixed conflict and compile problem. Local `make test` succeeds with the limited test set without the postgres database and I can not get the others to work with neither docker nor the test database outside docker. Travis tests state

```
    #   Failed test 'Upgrads not blocked'
    #   at ./t/api/02-iso.t line 713.
    #     Structures begin differing at:
    #          $got->{ha_alpha_node02_upgraded} = undef
    #     $expected->{ha_alpha_node02_upgraded} = 'ha_alpha_node01_upgrade'
    # Looks like you failed 1 test of 4.
#   Failed test 'Catch blocked_by cycles'
#   at ./t/api/02-iso.t line 725.
```

I don't know what "Upgrads" are ;)

**Comment by coolo** [comment]
yeah - good thing we have that test case.

Perhaps you rereview the 'fixed conflict' part :)

**Comment by okurz** [comment]
fixed fix and typo

**Comment by okurz** [comment]
could you please look into this PR and tell me when you see something missing? All checks have passed but it seems the coverage was only collected on an older state. Is it still acceptable or should I change something?

**Pull request merged**